### PR TITLE
ConnectionUtil refactoring (and splitting out a NetworkAttributesAppe…

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**
…nder)

### CODE REVIEW

File: splunk-otel-android/src/main/java/com/splunk/rum/RumInitializer.java [hunk 1]
```diff
@@ -116,15 +116,16 @@ SplunkRum initialize(ConnectionUtil.Factory connectionUtilFactory, Looper mainLo
...
-                            new RumAttributeAppender(visibleScreenTracker, connectionUtil);
+                    ScreenAttributesAppender screenAttributesAppender =
+                            new ScreenAttributesAppender(visibleScreenTracker);
                     initializationEvents.add(
                             new RumInitializer.InitializationEvent(
                                     "attributeAppenderInitialized", timingClock.now()));
 
                     tracerProviderBuilder
-                            .addSpanProcessor(attributeAppender)
+                            .addSpanProcessor(new NetworkAttributesAppender(connectionUtil))
```
Comment (breedx-splk) [inline splunk-otel-android/src/main/java/com/splunk/rum/RumInitializer.java:126]: The others aren't created in-line like this, maybe just make a local var for consistency?

File: splunk-otel-android/src/main/java/com/splunk/rum/ConnectionUtil.java [hunk 2]
```diff
@@ -102,26 +103,19 @@ CurrentNetwork getActiveNetwork() {
...
-                    ConnectionUtil.this.connectionStateListener;
-            if (connectionStateListener != null) {
-                connectionStateListener.onAvailable(true, activeNetwork);
-                Log.d(
-                        SplunkRum.LOG_TAG,
-                        "  onAvailable: isConnected:"
-                                + isOnline()
-                                + ", activeNetwork: "
-                                + activeNetwork);
+            Log.d(SplunkRum.LOG_TAG, "  onLost: activeNetwork=" + activeNetwork);
```
Comment (breedx-splk) [inline splunk-otel-android/src/main/java/com/splunk/rum/ConnectionUtil.java:115]: Whoops. 😉 

```suggestion
            Log.d(SplunkRum.LOG_TAG, "  onAvailable: activeNetwork=" + activeNetwork);
```

File: splunk-otel-android/src/main/java/com/splunk/rum/ConnectionUtil.java [hunk 3]
```diff
@@ -134,13 +128,10 @@ public void onLost(@NonNull Network network) {
...
-            if (connectionStateListener != null) {
-                connectionStateListener.onAvailable(false, activeNetwork);
-                Log.d(
-                        SplunkRum.LOG_TAG,
-                        "  onLost: isConnected:" + false + ", activeNetwork: " + activeNetwork);
+            Log.d(SplunkRum.LOG_TAG, "  onLost: activeNetwork=" + activeNetwork);
+
+            for (NetworkChangeListener listener : listeners) {
+                listener.onNetworkChange(activeNetwork);
             }
```
Comment (breedx-splk) [inline splunk-otel-android/src/main/java/com/splunk/rum/ConnectionUtil.java:135]: Duplicate dispatch loop. Factor out a method.

File: splunk-otel-android/src/main/java/com/splunk/rum/ConnectionUtil.java [hunk 4]
```diff
@@ -102,26 +103,19 @@ CurrentNetwork getActiveNetwork() {
...
-            Log.d(SplunkRum.LOG_TAG, "onAvailable: ");
             CurrentNetwork activeNetwork = refreshNetworkStatus();
-            ConnectionStateListener connectionStateListener =
-                    ConnectionUtil.this.connectionStateListener;
-            if (connectionStateListener != null) {
-                connectionStateListener.onAvailable(true, activeNetwork);
-                Log.d(
-                        SplunkRum.LOG_TAG,
-                        "  onAvailable: isConnected:"
-                                + isOnline()
```
Comment (breedx-splk) [inline splunk-otel-android/src/main/java/com/splunk/rum/ConnectionUtil.java:122]: With this removal, is `isOnline()` still used? I couldn't really find usages of it anymore but might have just missed it. Probably just delete it.

**Review by breedx-splk — CHANGES_REQUESTED** [review]
Looking good! A few small asks. 🤘🏻

File: splunk-otel-android/src/main/java/com/splunk/rum/ConnectionUtil.java (hunk 2 above)
Comment (mateuszrzeszutek) [reply splunk-otel-android/src/main/java/com/splunk/rum/ConnectionUtil.java:115]: 🙈

File: splunk-otel-android/src/main/java/com/splunk/rum/ConnectionUtil.java (hunk 4 above)
Comment (mateuszrzeszutek) [reply splunk-otel-android/src/main/java/com/splunk/rum/ConnectionUtil.java:122]: Nope, I missed that one -- will remove 👍

**Review by breedx-splk — APPROVED** [review]

**Pull request merged**